### PR TITLE
Realtime resolution update

### PR DESCRIPTION
**Pull request by Jnegrier**

### CODE REVIEW

File: components/templates/ListOfSuites.tsx [hunk 1]
```diff
@@ -5,7 +5,8 @@ import {
   showTestStats,
   showResolutionText,
   TestExpanded,
-} from "../../components/templates"
+  ListOfTests
+} from "."
```
Comment (opishcheiko) [inline components/templates/ListOfSuites.tsx:9]: what's up with those imports?

**Review by opishcheiko — COMMENTED** [review]
thanks for separating it into component!
Is it going to help with websockets? Like since it's a component - it's easier to refresh it?

**Comment by Jnegrier** [comment]
> thanks for separating it into component!
> Is it going to help with websockets? Like since it's a component - it's easier to refresh it?

Yep 😄 cause thanks to this, we set a state for each test, and since we are going to receive a test_id from the websocket, we can also filter the id directly between these tests, so we avoid to iterate across test_runs and test_suites :)

**Comment by Jnegrier** [comment]
> ! = 1 ? 🤔 Is it `not set` or something

For some reason, it was set as 1 always, so the piece of code to render the resolution based on the main test eas not working, it was always rendering it based on test_history

**Comment by Jnegrier** [comment]
Feel free to take a look 👍

File: components/templates/ListOfTests.tsx [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+  let filteredTest = tests.find(
+    test => test.test_id === testResolution.test_id);
+    if (filteredTest){
+      let testIndex = tests.indexOf(filteredTest);
+      if (filteredTest.test_history_id == testResolution.test_history_id){
+        filteredTest.test_history_resolution = testResolution.test_history_resolution
+        updateTest(testIndex, filteredTest)
+      } else if (filteredTest.test_history_resolution == 1 || !filteredTest.test_history_resolution){
+        filteredTest.test_resolution = testResolution.test_resolution
+        updateTest(testIndex, filteredTest)
```
Comment (opishcheiko) [inline components/templates/ListOfTests.tsx:112]: logic in this function seems kinda complicated, maybe we can simplify it later. Do you mind putting a small comment explaining those ifs, pleeeease? 🙏

File: pages/index.tsx [hunk 3]
```diff
@@ -225,6 +225,35 @@ export interface SuiteAndTest {
...
+    hours: number
+    minutes: number
+    months: number
+    seconds: number
+    years: number
+  }
+  test_history_id: number
+  test_id: number
+  name: string
+  test_history_resolution: number
```
Comment (opishcheiko) [inline pages/index.tsx:240]: Juan, this `test_history_resolution` is a string in SuiteAndTest interface, so maybe change both, or just leave it as string for simplifying? Why number?

File: components/templates/Test.tsx [hunk 4]
```diff
@@ -75,21 +73,23 @@ export const TestExpanded = function(props: TestProps) {
...
-  const { data, error } = useSWR(
-    `${process.env.publicDeltaCore}/api/v1/check_if_more_than_five_failed_in_the_last_ten_runs/test_id/${children.test_id}`,
-    fetcher
-  )
+  //TODO: replace this with using test.is_flaky property of test
+  
+  // const { data, error } = useSWR(
+  //   `${process.env.publicDeltaCore}/api/v1/check_if_more_than_five_failed_in_the_last_ten_runs/test_id/${children.test_id}`,
+  //   fetcher
+  // )
```
Comment (opishcheiko) [inline components/templates/Test.tsx:81]: commenting out as I forgot there is a breaking change on the backend

Comment (Jnegrier) [reply components/templates/Test.tsx:81]: Thanks!

File: pages/index.tsx (hunk 3 above)
Comment (Jnegrier) [reply pages/index.tsx:240]: I got this error if I keep it as a string, also I see that in [showResolution.js](https://github.com/delta-reporter/delta-frontend/blob/realtime-resolution-upt/components/templates/showResolution.js) is always used as a number as well, is returned as a string from the backend?

Screenshot 2020-09-24 at 10 38 52

File: components/templates/ListOfTests.tsx (hunk 2 above)
Comment (Jnegrier) [reply components/templates/ListOfTests.tsx:119]: Cool comments just added 👍

File: pages/index.tsx (hunk 3 above)
Comment (opishcheiko) [reply pages/index.tsx:240]: it's probably a number... I don't know, either compare it to `"1"` or change it to be a number everywhere :) 
Sorry, it's messy

**Review by opishcheiko — APPROVED** [review]
all good, thanks for fixing it!👍 
Just that bit with resolution being either a number everywhere, or a string, we should change it either/either

File: components/templates/ListOfTests.tsx (hunk 2 above)
Comment (opishcheiko) [reply components/templates/ListOfTests.tsx:119]: still don't fully understand what those ifs do. sorry 😢 
So every time we update `test_history` resolution, we need to update `test_resolution` as well, right?

**Comment by Jnegrier** [comment]
> all good, thanks for fixing it!👍
> Just that bit with resolution being either a number everywhere, or a string, we should change it either/either

Done! is now consistent everywhere

File: components/templates/ListOfTests.tsx (hunk 2 above)
Comment (Jnegrier) [reply components/templates/ListOfTests.tsx:119]: Yep, that's correct, but if the test the user has open, already has a `test_history_resolution`, we don't update  `test_resolution`, as there is already a specific resolution for the test on that specific run.

Probably we need to put more focus into differentiating `test_resolution` and `test_history_resolution`

**Pull request merged**